_overviews/scala3-scaladoc/search-engine.md

```diff
@@ -0,0 +1,86 @@
+---
+
+Searching for functions by their symbolic names can be time-consuming.
+That is why the new scaladoc allows searching for methods and fields by their types.
+
+
+Consider the following extension method definition:
+```
+extension [T](arr: IArray[T]) def span(p: T => Boolean): (IArray[T], IArray[T]) = ...
+```
+Instead of searching for `span` we can also search for `IArray[A] => (A => Boolean) => (IArray[A], IArray[A])`.
+
+To use this feature, type the signature of the member you are looking for in the scaladoc searchbar. This is how it works:
+
+![]({{ site.baseurl }}/resources/images/scala3/scaladoc/inkuire-1.0.0-M2_js_flatMap.gif)
+
+This feature is provided by the [Inkuire](https://github.com/VirtusLab/Inkuire) search engine, which works for Scala 3 and Kotlin. To be up-to-date with the development of this feature, follow the [Inkuire repository](https://github.com/VirtusLab/Inkuire).
+
+## Example queries
+
+Some examples of queries with intended results:
+- `List[Int] => (Int => Long) => List[Long]` -> `map`
+- `Seq[A] => (A => B) => Seq[B]` -> `map`
+- `(A, B) => A` -> `_1`
+- `Set[Long] => Long => Boolean` -> `contains`
+- `Int => Long => Int` -> `const`
+- `String => Int => Char` -> `apply`
+- `(Int & Float) => (String | Double)` -> `toDouble`, `toString`
+- `F[A] => Int` -> `length`
+
+## Query syntax
+
+In order for a scaladoc searchbar query to be searched using Inkuire instead of the default search engine, the query has to contain the `=>` character sequence. 
+
+Accepted input is similar to a curried function signature in Scala 3. With some differences:
+- AndTypes, OrTypes and Functions have to be enclosed in parentheses e.g. `(Int & Any) => String`
+- fields and parameterless methods can be found by preceding their type with `=>`, e.g., `=> Int`
+- A wildcard `_` can be used to indicate that we want to match any type in a given place e.g. `Long => Double => _`
+- Types in the form of single letter e.g. `A` or a letter with a digit `X1` are automatically assumed to be type variables
+- Other type variables can be declared just like in polymorphic functions e.g. `[AVariable, AlsoAVariable] => AVariable => AlsoAVariable => AVariable` 
+
+## How it works
+
+Inkuire works as a JavaScript worker in the browser thanks to the power of [ScalaJS](https://www.scala-js.org/).
+
+To enable Inkuire when running scaladoc, add the flag `-Ygenerate-inkuire`. By adding this flag two files are generated:
+- `inkuire-db.json` - this is the file containing all the searchable declarations from the currently documented project in a format readable to the Inkuire search engine.
+- `inkuire-config.json` - this file contains the locations of the database files that should be searchable from the documentation of the current project. By default, it will be generated with the location of the local db file as well as the default implied locations of database files in [External mappings](/scala3/guides/scaladoc/settings.html#-external-mappings).
```

Comment:
What do you think of moving this section last? It seems to me that “Query syntax” and “What to expect?” both provide information about how to use the feature, so it makes sense to group them (imho).

Reply:
Sure, done